pkg/rc/replication_controller.go

```diff
@@ -194,6 +186,42 @@ func (rc *replicationController) WatchDesires(quit <-chan struct{}) <-chan error
 	// When seeing any changes, try to meet them.
 	// If meeting produces any error, send it on the output error channel.
 	go func() {
+		defer func() {
+			rc.nodeTransferMu.Lock()
+			if rc.nodeTransfer.quit != nil {
+				rc.nodeTransfer.rollbackReason = "RC farm quitting"
+				close(rc.nodeTransfer.quit)
+			}
+			rc.nodeTransferMu.Unlock()
+		}()
+		// Fetch the RC's status to find out of a node transfer is underway
```

Comment:
Is this necessary? If a node transfer is underway, there will be an ineligible node, so in `meetDesires` this rc will call transferNodes which will call `rcStatusStore.Get` anyway

Reply:
When `scheduleNewTransferNodes` fails, for instance, it resets `rc.nodeTransfer` which is why we've been relying on that`rcStatusStore.Get` at the beginning of `transferNodes`

Reply:
It's necessary because now if a node transfer is underway, we'll have an extra node (since both old and new nodes are labeled), and we need to be able to prevent `removePods()` from removing anything.

I could replace this fetch with one in `removePods()`, but it's probably cleaner to initialize at startup and then anywhere in this file can trust `rc.nodeTransfer` as long as the mutex is held

Reply:
Understood. But as a followup, what then is stopping `removePods()` from getting rid of something after `rc.nodeTransfer` is wiped after a failed `scheduleNewTransferNodes`

Reply:
Good catch! We should not clear the data when a failure happens, so we'll retry with the same node in a few minutes.